TEP003_commit_message_guidelines.rst

```diff
@@ -0,0 +1,136 @@
+    
+* If there is more than one file touched in different ways and the one-line description isn't enough to describe all changes, the commit message needs a full-body description. The full-body description would consist of a description of the changes and a list of changes to individual files. The description of changes should answer the following three questions:
+
+  * Why is this change necessary?
+  * How does it address the issue?
+  * What side effects does this change have?
+
+If the changes are simple and/or the answers to the above questions are obvious from the context then you can ommit it.
+
+  Individual files you touched get their changes separately enumerated. If there a particular C or C++ function you changed, the general format is:
+
+  * file.py (class1.function1): Make change foo. 
+     (class2.function2): Make change bar.
+  * Note the newline for describing the change to a different function. General GNU style guidelines can be followed here, so that similar changes to different files can be grouped.
+  * For style points, please also write "New function" instead of "Added function" or "Return retval" instead of "Changed to return retval".
+  * And please, never "Fixed bug" or similar. That doesn't add any specific information about what was changed.
+
+* The changelog info in the commit message should say what changed, not why. If you need to explain why, that info should probably go in comments in the code.
+
+Commit Message Examples
+-----------------------
+
+Standard commit messages:
+
+Example 1:
+
+::
+
+   ensure PEP8 is followed in our code (issue #427)
+
+   * datahandler.py, docs/conf.py, interface.py, widgets.py: changes to make sure that PEP8 is followed
+
+Example 2:
+
+::
+
+   allow user to specify the number of threads in the configuration file
+
+   * data/tardis_config_definition.yml: added nthreads attribute under montecarlo
+   * model.py (Radial1DModel.simulate): pass the required number of threads to the C part of montecarlo routines
+
+Example 3:
+
+::
+
+   added --with-vpacket-logging build option (bug #428)
+
+   * setup.py: added --with-vpacket-logging flag for install, build and develop commands
+   * montecarlo/setup_package.py: define macro WITH_VPACKET_LOGGING if --with-vpacket-logging flag is used
+
+
+For short and very simple commit messages under 80 characters:
+
+Example 1:
+
+::
+
+   tests/test_atomic.py: added a test for atomic.read_levels_data
+
+Example 2:
+
+::
+
+   montecarlo/src/cmontecarlo.c: fix indentation
+
+Maintenance or docs commits, no actual code changed:
+
+Example 1:
+
+::
+
+   maint: updated .travis.yml file to do a rehash before executing conda
+
+
+Implementation
+==============
+
+Link to this TEP in the documentation for Contributors.
+
```

Comment:
I don't know where we should put this - but I think these templates are a great idea: https://github.com/blog/2111-issue-and-pull-request-templates